### PR TITLE
feat: [FFM-6890]: Improving audit wording for add rule

### PR DESCRIPTION
**Pull request by laura-mannus**
### Summary

**AS A** FF admin user
**I WANT** to see what rule was added in plain English
**SO THAT** I can understand how this impacts targeting without needing to understand the YAML diff

I have updated the add rule wording for both target and target groups to read more clearly. There was a limitation for the remove rule on target groups as we don't have the data for that. It will remain "Rule removed" however the log for Target is more verbose.

#### Screenshots

Before

image

After

image

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/75-cf/strings/strings.en.yaml [hunk 1]
```diff
@@ -391,10 +391,11 @@ auditLogs:
       bucketBy: Default serve updated for bucket by '{{bucketBy}}'
       variation: Default serve updated with variation '{{variation}}'
     addTargetsToVariationTargetMap: Target '{{target}}' was added for Variation '{{variation}}'
+    addTargetToFlagsVariationTargetMap: Target '{{target}}' was added for Variation '{{variation}}'
```
Comment (laura-mannus) [inline src/modules/75-cf/strings/strings.en.yaml:394]: is there a better way to achieve this than having two with the same value? I tried using `addTargetsToVariationTargetMap ` for both cases but this felt incorrect.

**Review by crystal-fong — DISMISSED** [review]
Should this be reviewed first, then if it's all good apply the changes into https://github.com/wings-software/ff-ui/pull/60/files ?

File: src/modules/75-cf/strings/strings.en.yaml (hunk 1 above)
Comment (knagurski) [reply src/modules/75-cf/strings/strings.en.yaml:394]: I *think* this could do the same job without duplication.
```suggestion
    addTargetToFlagsVariationTargetMap: {{ $.cf.auditLogs.events.addTargetsToVariationTargetMap }}
```

Not very pretty though. You could just use the same string id for both, but change it to be a little more generic... maybe `targetAddedForVariation`

File: src/modules/75-cf/components/AuditLogs/AuditLogsUtils.ts [hunk 2]
```diff
@@ -47,9 +46,11 @@ export function translateEvents(
 
       case 'addRule':
         message = getString('cf.auditLogs.events.addRule', {
-          clauses: Parameters?.clauses?.map((clause: Record<string, string>) => clause?.attribute || '').join(', ')
+          targetGroup: Parameters?.clauses?.map((clause: Record<string, string>) => clause?.values || '').join(', '),
+          variation: Parameters.variation
```
Comment (knagurski) [inline src/modules/75-cf/components/AuditLogs/AuditLogsUtils.ts:50]: I'm not sure this will work for a Percentage Rollout. In their case `Parameters.variation` won't exist and `Parameters.serve` will be in its place. 
Screenshot 2023-03-02 at 11 27 31

Comment (laura-mannus) [reply src/modules/75-cf/components/AuditLogs/AuditLogsUtils.ts]: I added some logic in there, what do you think of?
image

Should the wording need updated as well?

File: src/modules/75-cf/strings/strings.en.yaml (hunk 1 above)
Comment (laura-mannus) [reply src/modules/75-cf/strings/strings.en.yaml]: Have used the latter suggestion

Comment (knagurski) [reply src/modules/75-cf/strings/strings.en.yaml]: Yeah, I think that's the better idea. Referencing is good in some situations, but it feels kinda forced here.

File: src/modules/75-cf/components/AuditLogs/AuditLogsUtils.ts (hunk 2 above)
Comment (laura-mannus) [reply src/modules/75-cf/components/AuditLogs/AuditLogsUtils.ts]: Looks like this now 
image

File: src/modules/75-cf/components/AuditLogs/AuditLogsUtils.ts [hunk 3]
```diff
@@ -46,10 +45,19 @@ export function translateEvents(
...
+        Parameters.variation
+          ? (message = getString('cf.auditLogs.events.addRule', {
+              targetGroup: Parameters?.clauses
+                ?.map((clause: Record<string, string>) => clause?.values || '')
+                .join(', '),
+              variation: Parameters.variation
+            }))
+          : (message = getString('cf.auditLogs.events.addPercentageRollout', {
+              targetGroup: Parameters?.clauses?.map((clause: Record<string, string>) => clause?.values || '').join(', ')
+            }))
```
Comment (knagurski) [inline src/modules/75-cf/components/AuditLogs/AuditLogsUtils.ts:57]: Generally avoid assignments in ternaries... they're not super obvious and are usually better written as standard `if`/`else` statements
```suggestion
        if (Parameters.variation) {
          message = getString('cf.auditLogs.events.addRule', {
            targetGroup: Parameters?.clauses
              ?.map((clause: Record<string, string>) => clause?.values || '')
              .join(', '),
            variation: Parameters.variation
          })
        } else {
          message = getString('cf.auditLogs.events.addPercentageRollout', {
            targetGroup: Parameters?.clauses?.map((clause: Record<string, string>) => clause?.values || '').join(', ')
          })
        }
```

Or you could assign the output of the ternary
```suggestion
        message = Parameters.variation
          ? getString('cf.auditLogs.events.addRule', {
              targetGroup: Parameters?.clauses
                ?.map((clause: Record<string, string>) => clause?.values || '')
                .join(', '),
              variation: Parameters.variation
            })
          : getString('cf.auditLogs.events.addPercentageRollout', {
              targetGroup: Parameters?.clauses?.map((clause: Record<string, string>) => clause?.values || '').join(', ')
            })
```

**Comment by knagurski** [comment]
Great work, just that ternary thing and I think we're good to go!

File: src/modules/75-cf/components/AuditLogs/AuditLogsUtils.ts (hunk 3 above)
Comment (laura-mannus) [reply src/modules/75-cf/components/AuditLogs/AuditLogsUtils.ts]: Ok, I prefer the tidiness of the ternary so I have moved the assignment to the output 👍

**Review by knagurski — APPROVED** [review]
🔥 great work

**Review by crystal-fong — APPROVED** [review]
nice work 🦖
